### PR TITLE
made public keys a compile time setting/constant

### PR DESCRIPTION
**Pull request by ChristianKienle**
## Description
This PR makes our package verification more flexible.

## Checklist

* [x] This PR does not change text that hasn't been signed off with the RKI. Have a look at the pinned issue [440](https://github.com/corona-warn-app/cwa-app-ios/issues)
* [x] Test your changes as thoroughly as possible before you commit them. Preferably, automate your test by unit/integration tests.
* [x] If this PR comes from a fork, please [Allow edits from maintainers](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork)
* [x] Set a speaking title. Format: {task_name} (closes #{issue_number}). For example: Use logger (closes # 41)
* [x] [Link your Pull Request to an issue](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) - Pull Requests that are not linked to an issue with you as an assignee will be closed.
* [x] Create Work In Progress [WIP] pull requests only if you need clarification or an explicit review before you can continue your work item.
* [x] Make sure that your PR is not introducing _unnecessary_ reformatting (e.g., introduced by on-save hooks in your IDE)

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Workers/PublicKeyStore.swift [hunk 1]
```diff
@@ -29,26 +29,52 @@ enum KeyError: Error {
...
+		switch self {
+		case .production: return "c7DEstcUIRcyk35OYDJ95/hTg3UVhsaDXKT0zK7NhHPXoyzipEnOp3GyNXDVpaPi3cAfQmxeuFMZAIX2+6A5Xg=="
+		case .development: return "3BYTxr2HuJYQG+d7Ezu6KS8GEbFkiEvyJFg0j+C839gTjT6j7Ho0EXXZ/a07ZfvKcC2cmc1SunsrqU9Jov1J5Q=="
+		}
 	}
 }
+
+typealias PublicKeyProvider = () -> P256.Signing.PublicKey
+typealias PublicKeyFromStringProvider = (StaticString) -> PublicKeyProvider
+typealias PublicKeyProviderFromActiveCompilationConditions = () -> PublicKeyProvider
```
Comment (ArturFriesen) [inline src/xcode/ENA/ENA/Source/Workers/PublicKeyStore.swift:63]: I think this typealias is not used.

**Review by inf2381 — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Workers/PublicKeyStore.swift [hunk 2]
```diff
@@ -29,26 +29,52 @@ enum KeyError: Error {
...
+	var stringRepresentation: StaticString {
+		switch self {
+		case .production: return "c7DEstcUIRcyk35OYDJ95/hTg3UVhsaDXKT0zK7NhHPXoyzipEnOp3GyNXDVpaPi3cAfQmxeuFMZAIX2+6A5Xg=="
+		case .development: return "3BYTxr2HuJYQG+d7Ezu6KS8GEbFkiEvyJFg0j+C839gTjT6j7Ho0EXXZ/a07ZfvKcC2cmc1SunsrqU9Jov1J5Q=="
+		}
 	}
 }
+
+typealias PublicKeyProvider = () -> P256.Signing.PublicKey
+typealias PublicKeyFromStringProvider = (StaticString) -> PublicKeyProvider
```
Comment (ArturFriesen) [inline src/xcode/ENA/ENA/Source/Workers/PublicKeyStore.swift:62]: I think naming should be: "PublicKeyProviderFromStringProvider"

File: COMPILATION_CONDITIONS.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
+# Compilation Conditions
+
+Our code uses several (custom) [ compilation conditions](https://help.apple.com/xcode/mac/11.4/#/itcaec37c2a6) (search for `SWIFT_ACTIVE_COMPILATION_CONDITIONS`).
+
+| Compilation Condition | Description |
+|---|---|
+| `RELEASE` | `true` if the app was built in release-mode. We mainly use this flag to disable certain features (developer menu, logging, …) for production. If you want to disable a certain feature for production you should use `#if !RELEASE` as opposed to `#if DEBUG`. |
+| `UITESTING` | `true` is the app was built in order to execute UI tests. We mainly use this flag to have slightly different code paths which is required when generating screenshots or testing certain edge conditions.  |
+| `COMMUNITY` | `true` is the *Community* build configuration/schema has been selected. The *Community* configuration enables other developers to run the app in Simulator. |
+| `USE_DEV_PK_FOR_SIG_VERIFICATION` | `true` if the app should use the dev-public keys for the signature verfification. This is only set for testig and during development. In case this is set mistakenly to `true` the app simply stops working. |
```
Comment (ArturFriesen) [inline COMPILATION_CONDITIONS.md:10]: Typo: "testig"

**Review by marcussc — APPROVED** [review]
